File: backend-inmocasas/backend-inmocasas/api/models/Usuario.js
/**
 * Usuario.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    nombre:{
      type: 'string',
      required:true

    },
    apellido:{
      type: 'string',
      required:true

    },
    telefono:{
      type:'string',
      required:true,
      maxLength:10,
      minLength:10

    },
    nombreUsuario: {
      type: 'string',
      required:true

    },
    email:{
      type:'string',
      isEmail:true,
      required:true

    },
    contrasenia:{
      type:'string',
      required:true
    },
    rolesDeUsuario: {
      collection: 'rolUsuario',
      via: 'fkUsuario'
    },
    publicacionesDeUsuario: {
      collection: 'publicacion',
      via: 'fkUsuario'
    }


  },

};

File: Frontend-inmocasas/app/src/main/java/com/example/inmocasas/dto/Usuario.kt
package com.example.inmocasas.dto

import java.util.*
import kotlin.collections.ArrayList

class Usuario(
    var id: Int,
    var createdAt: Long,
    var updatedAt: Long,
    var nombre: String,
    var apellido:String,
    var telefono:String,
    var nombreUsuario:String,
    var email:String,
    var contrasenia:String,
    var rolesDeUsuario:ArrayList<RolUsuario>,
    var publicacionesDeUsuario:ArrayList<Publicacion>
) {
    var fechaCreacion: Date
    var fechaActualizacion: Date

    init {
        fechaCreacion = Date(createdAt)
        fechaActualizacion = Date(updatedAt)
    }


}